linux/devdoc/async_socket_linux_requirements.md

```diff
@@ -275,62 +233,58 @@ MOCKABLE_FUNCTION(, int, async_socket_receive_async, ASYNC_SOCKET_HANDLE, async_
 
 **SRS_ASYNC_SOCKET_LINUX_11_074: [** The context shall also allocate enough memory to keep an array of `buffer_count` items. **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_075: [** `async_socket_receive_async` shall add the socket in the epoll system by calling `epoll_ctl` with `EPOLL_CTL_MOD` **]**
-
-**SRS_ASYNC_SOCKET_LINUX_11_076: [** If the `epoll_ctl` call fails with `ENOENT`, `async_socket_send_async` shall call `epoll_ctl` again with `EPOLL_CTL_ADD`. **]**
-
 **SRS_ASYNC_SOCKET_LINUX_11_077: [** On success, `async_socket_receive_async` shall return 0. **]**
 
 **SRS_ASYNC_SOCKET_LINUX_11_078: [** If any error occurs, `async_socket_receive_async` shall fail and return a non-zero value. **]**
 
-### thread_worker_func
+### event_complete_callback
 
 ```c
-static int thread_worker_func(void* parameter)
+static void event_complete_callback(void* context, COMPLETION_PORT_EPOLL_ACTION action)
```

Comment:
>event_complete_callback [](http://example.com/codeflow?start=12&length=23)

I do have to ask annoying questions here... but...

for the same epoll, can these event_complete_callback originate on different threads?

In the case when a connection is severed, will be events queued in the epoll be served:
1) in the order that they have been queued
2) in the reverse order they have been queued.
3) in a random order.

This might be important, and should be stated if these callbacks come from different threads to the user land or 1 unique thread and the order (if any). Users might care.

Reply:
Good point, if the connection is severed the events will be serviced in the order they are queued.  I'll add that to the doc as a note.